discord/message_handlers.go

```diff
@@ -42,6 +42,179 @@ func (guild *GuildState) handleGameStartMessage(s *discordgo.Session, m *discord
 	guild.GameStateMsg.AddReaction(s, "❌")
 }
 
+func (guildState *GuildState) createPrivateMapMessage(s *discordgo.Session, m *discordgo.MessageCreate, channels []TrackingChannel) {
+	mapUsers(s, m, channels, guildState)
+	createMessage(s, guildState)
+}
+
+// START - createPrivateMapMessage Helper Methods
+func mapUsers(s *discordgo.Session, m *discordgo.MessageCreate, channels []TrackingChannel, guildState *GuildState) {
+
+	var idUsernameMap = populateIdUsernameMap(s, m.GuildID, channels, guildState);
+
+	guildState.PrivateStateMsg.idUsernameMap = idUsernameMap;
+}
+
+func populateIdUsernameMap(s *discordgo.Session, guildId string, channels []TrackingChannel, guildState *GuildState) map[string]string{
```

Comment:
Not sure I understand the purpose of this function. This seems really similar to `checkCacheAndAddUser` from guild.go, but perhaps I'm wrong.

Reply:
Everytime the .au new or the .au prompt command is used, this method is ran to loop through everyone inside of tracked channels. The 'printedUsers' map is just a way for me to determine who has been shown by the private prompt before. The idUsernameMap is a map of userID to their name. The reason I have this map is in case two users have the same name, I can add their discord tag to the end and update the map. I didn't understand how checkCacheAndAddUser worked and didn't want to mess anything up. But potentially, could I add players I've shown in the prompt to `guild.UserData`?

Reply:
yeah, so checkCacheAndAddUser is intended to loop over tracked channels and add those users to the guild.Userdata (add them to it as a "cache"), as you mentioned. So IMO the idUsernameMap is slightly redundant? I'll leave that aspect up to you (it sounds like it might be specific enough to hang around), I just wanted to make sure looping over the voicestates/channels was minimized here, if the checkCacheAndAddUser does some of that functionality already, but perhaps it doesn't quite fit your use case

Reply:
Now that you're explaining it, it does sound like those methods are almost identical. It'll be great to avoid redundant iterations! I think the part I'm confused apart right now with the cacheAndAddUser is the first part where it checks if they're in the cache:

```
//check and see if they're cached first
	for _, v := range g.Members {
                    // CloakingPotion Comment -> g.Members are just all members in the guild, right?
		if v.User.ID == userID {
                    // CloakingPotion Comment -> Doesn't this if statement just check if the user they're looking for is apart of the guild?

                   // CloakingPotion Comment -> This part down here the more confusing for me. If we find someone that matches, why do we still use 'MakeUserDataFromDiscordUser' and 'AddFullUser'. Wouldn't we do something closer to 'return guild.UserData[userId]'?
			user := game.MakeUserDataFromDiscordUser(v.User, v.Nick)
			guild.UserData.AddFullUser(user)
			return user, true
		}
	}
```

Reply:
I would push my current code for reference but there are a few errors and I haven't tested it quite yet